Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputSystem.cs

```diff
@@ -242,18 +242,24 @@ public override void Reset()
         /// <inheritdoc />
         public override void Disable()
         {
-            GazeProvider = null;
-
-            if (!Application.isPlaying)
+            // Input System adds a gaze provider component on the main camera, which needs to be removed when the input system is disabled/removed.
+            // Otherwise the component would keep references to dead objects.
+            // Unity's way to remove component is to destroy it.
+            if (GazeProvider != null)
             {
-                var component = CameraCache.Main.GetComponent<IMixedRealityGazeProvider>() as Component;
-
-                if (component != null)
+                if (Application.isPlaying)
+                {
+                    UnityEngine.Object.Destroy(GazeProvider as Component);
```

Comment:
This component should be added at edit time, so it seems strange to destroy it at runtime.
Especially in `Disable`, which is the inverse of `Enable`, not `Initialize` (which is where the gaze provider is added).

Reply:
This component is added in both Edit mode and Runtime mode. I've added destroying in runtime case, as destroy in Edit mode has already been here before. 
Part about `Initialize` is true, but the old behavior was to set GazeProvider to null in `Disable`. It isn't set to anything in `Enable`, so input system won't work correctly unless `Reset` is called. I preserved this behavior.

Reply:
>I've added destroying in runtime case, as destroy in Edit mode has already been here before.

I had thought this was going to be a problem, not knowing whether the provider was added at runtime or edit time, but from some testing it looks like I was wrong and the editor preserves state well.

>but the old behavior was to set GazeProvider to null in Disable

Good catch. Feels like there's a bug in there.